example/ExampleMain.cpp

```diff
@@ -172,9 +176,7 @@ void exampleVisit(Module &module) {
         b.add<UnaryInstruction>([](VisitorNest &self, UnaryInstruction &inst) {
           *self.out << "visiting UnaryInstruction: " << inst << '\n';
         });
-        b.add<BinaryOperator>([](VisitorNest &self, BinaryOperator &inst) {
-          *self.out << "visiting BinaryOperator: " << inst << '\n';
-        });
+        b.add(&VisitorNest::visitBinaryOperator);
```

Comment:
To encourage the strongly encouraged pattern where a visitor is constructed once as a static variable, maybe the example should show how that works?

Reply:
There's a tension between doing that and the fact that this visitor build process is actually configurable by an argument.

Reply:
If `setStrategy` returns the builder again, shouldn’t it work like this?
```c++
static VisitorBuilder<VisitorContainer> builder = VisitorBuilder<VisitorContainer>()
  .nest<VisitorNest>(…)
  .setStrategy(g_rpot ? VisitorStrategy::ReversePostOrder : VisitorStrategy::Default)
  .build();
```

Reply:
This might work in this example, but in general I'm not sure initializing static objects with mutable global state is something we'd recommend, given the possibility of that state changing and then having a stale static object?

You could of course use *different* builders for the different strategies which would avoid that (wrapped e.g. in a templated getter), but I'm not sure that's worth the effort?

Reply:
Ah sure, why not :)